### PR TITLE
[cleanup] Remove BinaryIndexView

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- Removing unnecessary enum following `CompiledScript` removal #17113 

## Test plan 

- CI

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: external-crates/move/crates/move-bytecode-source-map/src/source_map.rs [hunk 1]
```diff
@@ -490,46 +490,38 @@ impl SourceMap {
...
-        let module_ident = match view {
-            BinaryIndexedView::Module(..) => {
-                let module_handle = view.module_handle_at(ModuleHandleIndex::new(0));
-                let module_name = ModuleName(Symbol::from(
-                    view.identifier_at(module_handle.name).as_str(),
-                ));
-                let address = *view.address_identifier_at(module_handle.address);
-                ModuleIdent::new(module_name, address)
-            }
+    pub fn dummy_from_view(module: &CompiledModule, default_loc: Loc) -> Result<Self> {
```
Comment (dariorussi) [inline external-crates/move/crates/move-bytecode-source-map/src/source_map.rs:493]: `from_module`? don't know, whatever.... :)

File: external-crates/move/crates/move-bytecode-verifier/src/dependencies.rs [hunk 2]
```diff
@@ -498,15 +493,16 @@ fn verify_all_script_visibility_usage(context: &Context) -> PartialVMResult<()>
         Some(s) => s,
     };
     debug_assert!(context.resolver.version() < VERSION_5);
-    match &context.resolver {
-        BinaryIndexedView::Module(m) => {
+    let m = context.resolver;
+    {
+        {
```
Comment (dariorussi) [inline external-crates/move/crates/move-bytecode-verifier/src/dependencies.rs:498]: is the double block just to make the diff cleaner? a bit confusing

File: external-crates/move/crates/move-bytecode-verifier/src/script_signature.rs [hunk 3]
```diff
@@ -86,7 +85,7 @@ fn verify_module_function_signature(
 ) -> VMResult<()> {
     let fdef = module.function_def_at(idx);
 
-    let resolver = &BinaryIndexedView::Module(module);
+    let resolver = module;
```
Comment (dariorussi) [inline external-crates/move/crates/move-bytecode-verifier/src/script_signature.rs:88]: some of those assignments are a bit funny but I hear you....

File: external-crates/move/crates/move-vm-runtime/src/loader.rs
Comment (dariorussi) [inline external-crates/move/crates/move-vm-runtime/src/loader.rs:1]: changes here are mostly formatting right?

File: external-crates/move/move-execution/v1/crates/move-bytecode-verifier/src/script_signature.rs
Comment (dariorussi) [inline external-crates/move/move-execution/v1/crates/move-bytecode-verifier/src/script_signature.rs:1]: what is this file?

**Review by dariorussi — APPROVED** [review]
amazing how much it cleans things up. And it seems very reasonable and straightforward.
are you going to remove `enum BinaryType` from the loader too?
I guess less critical and it's just a useless enum for now :)

File: external-crates/move/crates/move-bytecode-source-map/src/source_map.rs (hunk 1 above)
Comment (tnowacki) [reply external-crates/move/crates/move-bytecode-source-map/src/source_map.rs:493]: I might do a separate pass for these sorts of things. Trying to keep the diff clean as I've made one mistake already

File: external-crates/move/crates/move-vm-runtime/src/loader.rs
Comment (tnowacki) [reply external-crates/move/crates/move-vm-runtime/src/loader.rs:1]: Seems like it yeah

File: external-crates/move/move-execution/v1/crates/move-bytecode-verifier/src/script_signature.rs
Comment (tnowacki) [reply external-crates/move/move-execution/v1/crates/move-bytecode-verifier/src/script_signature.rs:1]: Not quite sure I understand the question lol

But assuming I understad anyway, it is a verifier pass for dealing with `public(script)` migration and `entry` functions

File: external-crates/move/crates/move-bytecode-source-map/src/source_map.rs (hunk 1 above)
Comment (dariorussi) [reply external-crates/move/crates/move-bytecode-source-map/src/source_map.rs:493]: sounds like a good plan!

File: external-crates/move/move-execution/v1/crates/move-bytecode-verifier/src/script_signature.rs
Comment (dariorussi) [reply external-crates/move/move-execution/v1/crates/move-bytecode-verifier/src/script_signature.rs:1]: oh so it's not about scripts

Comment (tnowacki) [reply external-crates/move/move-execution/v1/crates/move-bytecode-verifier/src/script_signature.rs:1]: It was at one point in time

**Pull request merged**